### PR TITLE
Lost

### PR DESCRIPTION
**Pull request by tsbolty**
### What this PR does (required):
- Added back some missing code in the SubmittedDesigns component and the Contest component as well as corresponding routes.

Specifically

- Missing Contest page and corresponding styling
- Missing route for "more info" contest button
- Missing route in app.tsx for "/contest" page
- Missing some code in front end helpers/contest.ts to get one contest
- Fixed typo in title element

### Screenshots / Videos (required):
None

### Any information needed to test this feature (required):
- Just use the app :)

### Any issues with the current functionality (optional):
- None

### CODE REVIEW

File: client/src/pages/Contest/Contest.tsx [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+      opacity: '0.8',
+    },
+  },
+})(() => null);
+
+export default function Contest(): JSX.Element {
+  const { id } = useParams() as { id: string };
+  const classes = useStyles();
+  const [value, setValue] = useState(0);
+  //const [contestSubmissions, setContestSubmissions] = useState<Submission[]>([]);
```
Comment (rajivtitus) [inline client/src/pages/Contest/Contest.tsx]: Please remove

File: client/src/pages/Contest/Contest.tsx [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+              <Grid item>
+                <Avatar alt="Profile Image" src={ProfilePic} className={classes.avatar}></Avatar>
+              </Grid>
+              <Grid item>
+                <Typography className={classes.user}>By {capitalize(loggedInUser.username)}</Typography>
+              </Grid>
+            </Grid>
+          </Grid>
+          <Grid item xs={2}>
+            <Link to={`/submit-design/${id}`} style={{ textDecoration: 'none' }}>
```
Comment (rajivtitus) [inline client/src/pages/Contest/Contest.tsx:158]: Link and Button can be combined here

Ref: https://stackoverflow.com/questions/51642532/how-to-make-a-material-ui-react-button-act-as-a-react-router-dom-link

File: client/src/pages/Discovery/Discovery.tsx [hunk 3]
```diff
@@ -18,120 +18,116 @@ import TableCell from '@material-ui/core/TableCell';
...
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(+event.target.value);
+    setPage(0);
+  };
 
-            setContests(sort)
+  const sortByHeader = (sortParam: Contest[] = contests) => {
+    if (contests) {
+      const sort = [...sortParam].sort((a: Contest, b: Contest) => {
+        console.log(a, b);
```
Comment (rajivtitus) [inline client/src/pages/Discovery/Discovery.tsx]: Please remove

File: client/src/pages/Contest/Contest.tsx [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        {value === props.index && (
+          <Container>
+            <Typography>{props.children}</Typography>
+          </Container>
+        )}
+      </div>
+    );
+  };
+
+  const capitalize = (s: string) => s && s[0].toUpperCase() + s.slice(1);
```
Comment (rajivtitus) [inline client/src/pages/Contest/Contest.tsx]: CSS has a text-transform property to capitalize strings, please use that instead :)

File: client/src/pages/Contest/Contest.tsx (hunk 1 above)
Comment (tsbolty) [reply client/src/pages/Contest/Contest.tsx]: This is fixed.

File: client/src/pages/Contest/Contest.tsx (hunk 2 above)
Comment (tsbolty) [reply client/src/pages/Contest/Contest.tsx:158]: This is fixed.

File: client/src/pages/Discovery/Discovery.tsx (hunk 3 above)
Comment (tsbolty) [reply client/src/pages/Discovery/Discovery.tsx]: This is fixed.

File: client/src/pages/Contest/Contest.tsx [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                  {contestSubmissions.prizeAmount ? contestSubmissions.prizeAmount : '$150'}
+                </Typography>
+              </Button>
+            </Typography>
+            <Grid direction="row" className={classes.grid} container>
+              <Grid item>
+                <Avatar alt="Profile Image" src={ProfilePic} className={classes.avatar}></Avatar>
+              </Grid>
+              <Grid item>
+                <Typography className={classes.user}>By {capitalize(loggedInUser.username)}</Typography>
```
Comment (rajivtitus) [inline client/src/pages/Contest/Contest.tsx]: Use this class to apply styles to the string, please see above for reference

File: client/src/helpers/APICalls/contest.ts [hunk 6]
```diff
@@ -4,43 +4,44 @@ import { Contest } from '../../interface/Contest';
...
+    credentials: 'include',
+  };
 
-    return await fetch('/contest', fetchData)
-        .then(data => data.json())
-        .catch(err => ({ error: { message: 'Can not connect to server' } }))
-}
+  return await fetch('/contest', fetchData)
+    .then((data) => data.json())
+    .catch((err) => ({ error: { message: 'Can not connect to server' } }));
```
Comment (rajivtitus) [inline client/src/helpers/APICalls/contest.ts]: I just randomly notice a typo here, 'Cannot' 😄

**Review by rajivtitus — APPROVED** [review]
LGTM! 
Please address comments before merging, maybe also add a more meaningful title to the PR (Refactor/Missing Code - name of components) 

Thank you!

File: client/src/pages/Contest/Contest.tsx (hunk 4 above)
Comment (tsbolty) [reply client/src/pages/Contest/Contest.tsx]: This is fixed.

File: client/src/pages/Contest/Contest.tsx (hunk 5 above)
Comment (tsbolty) [reply client/src/pages/Contest/Contest.tsx]: This is fixed.

File: client/src/helpers/APICalls/contest.ts (hunk 6 above)
Comment (tsbolty) [reply client/src/helpers/APICalls/contest.ts]: This is fixed.

**Pull request merged**